src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -98,6 +99,8 @@ private ILocalGrainDirectory Directory
 
         #region Implementation of IRuntimeClient
 
+        public IGrainReferenceRuntime GrainReferenceRuntime => this.grainReferenceRuntime ?? (this.grainReferenceRuntime = this.ServiceProvider.GetRequiredService<IGrainReferenceRuntime>());
```

Comment:
Yeah, I copied the other members in this class. I'll do a quick check to see if I can break the dependency cycle, otherwise, we already know this runtimeclient acts as a glorified strongly typed service locator, that we should eventually get rid off (at least the properties that do this)